data_validation/cli_tools.py

```diff
@@ -978,10 +989,9 @@ def _add_common_partition_arguments(optional_arguments, required_arguments):
         "--partition-num",
         "-pn",
         required=True,
-        help="Number of partitions/config files to generate",
-        type=int,
-        choices=range(1, 1001),
-        metavar="[1-1000]",
+        help="Number of partitions/config files to generate, a number from 1 to 10,000",
+        type=_check_no_partitions,
```

Comment:
Instead of having a typecheck function, you can use the metavar parameter of `add_argument()` to change the description: https://stackoverflow.com/questions/25295487/python-argparse-value-range-help-message-appearance

Reply:
Yes - however when you read the comment on that - metavar only helps with usage and help, but not with error - which is the problem I noticed - hence the link I posted in my first message.